### PR TITLE
Include plugins in static build

### PR DESCRIPTION
**Pull request by HeroCC**
fixes https://github.com/docker/for-linux/issues/1066

Docker documentation states that plugins like buildx and app are included in distributions as of v19.13 (https://github.com/docker/docker-ce-packaging/pull/309).   However, this doesn't include the static builds. Downstream, people expect these to be included and may be confused to learn that the static build doesn't include them: https://github.com/docker-library/docker/issues/156. Including the plugins would make the builds more consistent.

This is my first contribution to the docker project, so apologies if this breaks anything 😅

### CODE REVIEW

**Comment by HeroCC** [comment]
Hmm, is the CI working correctly? #414 has the same build error as I do. I'd like to make sure it actually builds before merging

**Comment by thaJeztah** [comment]
```
Step 24/49 : RUN go build -o /sbin/dockerd     -tags 'autogen apparmor seccomp selinux journald exclude_graphdriver_devicemapper'     -i     -buildmode=pie     -a -ldflags '-w'    github.com/docker/docker/cmd/dockerd
 ---> Running in 2d737193b473
# github.com/docker/docker/dockerversion
dockerversion/useragent.go:20:83: undefined: Version
dockerversion/useragent.go:22:87: undefined: GitCommit
The command '/bin/sh -c go build -o /sbin/dockerd     -tags 'autogen apparmor seccomp selinux journald exclude_graphdriver_devicemapper'     -i     -buildmode=pie     -a -ldflags '-w'    github.com/docker/docker/cmd/dockerd' returned a non-zero code: 2
Makefile:61: recipe for target 'image-linux' failed
```

Suspect that's due to upstream changes in https://github.com/moby/moby/pull/40180

Specifically this commit; https://github.com/moby/moby/pull/40180/commits/675b414f56063826457cbc8f0c165af67198bfa8

**Comment by HeroCC** [comment]
Thanks for fixing the upstream on this @thaJeztah! I rebased my branch and now the build appears to compile. When you get the chance, can you review it and let me know if everything looks good from your end?

**Comment by thaJeztah** [comment]
I just realise the "DCO check" is not enabled on this repository, and your commit is missing a DCO sign off; would you be able to amend your commit, and add a sign-off to the commit message? (see https://github.com/moby/moby/blob/master/CONTRIBUTING.md#sign-your-work)

Unfortunately, you can't do this from GitHub's web UI, so you'd have to do this from the command line;

If you have your name and e-mail (`user.name` and `user.email`) configured in `git config -l`, you can use the `-s` flag to do this automatically;

- clone your fork of the repository locally
- switch to the `patch-1` branch
- amend your commit message (`git commit -s --amend`)
    - this should add the `Signed-off-by:` line, but make sure it contains your _real_ name, otherwise edit it manually (see below)
- force push your branch to github (`git push -f origin`)

The `Signed-off-by` line should be the last line in your commit message, and look like:

```
Signed-off-by: Your Real Name <your-email@example.com>
```

**Comment by thaJeztah** [comment]
As a side-note: we should decide wether we want to include the plugins in the same tar.gz, or provide separate downloads for individual plugins (so that we can decouple releases of both), but I guess if we go that road, it would likely make sense to do the same for the `.deb` and `.rpm` packages (so not sure)

**Comment by HeroCC** [comment]
Just signed off and re-pushed. 

Regarding separating plugins, if they are separated would they still be included by default in the docker-ce package? Right now most docs say that buildx is [included on 19.03](https://docs.docker.com/buildx/working-with-buildx/#install), which is what I am trying to replicate for static releases w/ this PR. Would plugins be installed with something like docker-ce-cli-plugins? 

Personally, for now I'm good with them all being included together, but am not opposed to them being separated as long as it is consistent between packaging type.

**Comment by HeroCC** [comment]
Sorry for ping, but can I get a review on this? @thaJeztah

**Comment by iAnomaly** [comment]
@StefanScherer @russellcardullo @thaJeztah any chance you approve? Based on @tianon 's comment here https://github.com/docker-library/docker/issues/156#issuecomment-558410863 , the official Docker images are missing the "plugins" bundled with Docker CE as per the docs: https://github.com/docker/buildx/

It would be very helpful and avoid further confusion by bringing parity to the static builds and downstream official images by including these plugins. Thank you!

**Comment by Vhab** [comment]
@StefanScherer @russellcardullo @thaJeztah Let's not forget about this one.
This seems to be a blocker for getting buildx into the official docker image, which will greatly improve the ability to use buildx in CI and other contexts.

**Review by seemethere — APPROVED** [review]

**Review by duhruh — APPROVED** [review]

**Comment by LorbusChris** [comment]
Having `buildx` included would be hugely useful.

Is there anything missing here? bumping to hopefully get this merged in soon :)

**Comment by maniator** [comment]
> Having `buildx` included would be hugely useful.
> 
> Is there anything missing here? bumping to hopefully get this merged in soon :)

I created this image just for that purpose 😃 https://hub.docker.com/r/maniator/dind-buildx

**Comment by davidoram** [comment]
This would be super useful on my projects.  Hopefully this will be merged soon 🤞 👍

**Comment by korolkovas** [comment]
Up 👍 . Enabling build multi-platform images out of the box would be excellent.

**Comment by nappa32** [comment]
I would like this feature. 👍

**Comment by maniator** [comment]
See above @nappa32 @korolkovas @davidoram https://github.com/docker/docker-ce-packaging/pull/415#issuecomment-869271217

**Comment by ajhalili2006** [comment]
Will this PR ever be merged in the foreseeable future or is this abandoned PR left to stale?

**Comment by HeroCC** [comment]
Working on bringing this PR more up to date -- please excuse the temporary messy commit history. GitHub didn't rebase cleanly at all. Going through and rebasing it manually, and making the static plugins multiarch.

**Comment by daveisfera** [comment]
The use of `buildx` was [advertised again today](https://www.docker.com/blog/image-rebase-and-improved-remote-cache-support-in-new-buildkit/), so any chance we can get this merged?

**Pull request closed without merging**
